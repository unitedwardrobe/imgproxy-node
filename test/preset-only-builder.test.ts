import Imgproxy from '../src';

describe('preset-only-builder', () => {
  let imgproxy: Imgproxy;
  const imageUrl = 'https://example.com/image.jpg';

  beforeEach(async () => {
    imgproxy = new Imgproxy({
      baseUrl: 'https://example.com',
      encode: false,
      insecure: false,
      key: 'awesomeKey',
      salt: 'awesomeSalt',
    });
  });

  it('should work when no presets are passed', async () => {
    const builder = imgproxy.presetOnlyBuilder().preset();
    expect(builder.generateUrl(imageUrl)).toMatchSnapshot();
  });

  it('should work when presets are passed', async () => {
    const builder = imgproxy
      .presetOnlyBuilder()
      .preset('thumb_256', 'blur', 'center_crop');
    expect(builder.generateUrl(imageUrl)).toMatchSnapshot();
  });
});
