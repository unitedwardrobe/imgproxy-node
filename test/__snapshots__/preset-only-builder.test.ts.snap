// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset-only-builder should work when no presets are passed 1`] = `"https://example.com/pEvkDJzRVINGZbennLcmPMoNWObtxVBIWqUIErUOjbk/plain/https://example.com/image.jpg"`;

exports[`preset-only-builder should work when presets are passed 1`] = `"https://example.com/vrZVbs-iwrOWgQ-zmliLuRKMpzyAKpo4mODQhIcJ_lQ/thumb_256:blur:center_crop/plain/https://example.com/image.jpg"`;
